src/main/java/datawave/query/util/DateFrequencyValue.java

```diff
@@ -0,0 +1,159 @@
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * This class handles the serialization and deserialization of the Accumulo value in a record of the Datawave Metadata table that has a column family of "f" and
+ * a column qualifier that is prefixed with the string "compressed-" like "compressed-csv" for example. This is a class used to help compress the date and
+ * frequency values that are aggregated together to by the FrequencyTransformIterator and manipulated in the FrequencyFamilyCounter. The byte array is in
+ * regular expression format ((YEAR)(4BYTE-FREQUENCY){366}))* . Explained verbally a four byte representation of Year followed by by up to 366 (Leap year) 4
+ * byte holders for frequency values. The month and day of the frequency value is coded by the position in the array. There aren't any delimiters between years
+ * and frequencies which adds to the compression. Each Accumulo row for this "aggregated" frequency "map" would be 10 x ( 4 + 4 + (366 * 4) ) bytes long for a
+ * maximum length for a 10 year capture of 14720 bytes.
+ */
+
+public class DateFrequencyValue {
+    
+    private static final Logger log = LoggerFactory.getLogger(DateFrequencyValue.class);
+    
+    public static final int DAYS_IN_LEAP_YEAR = 366;
+    private static int NUM_YEAR_BYTES = 4;
+    private static int NUM_FREQUENCY_BYTES = DAYS_IN_LEAP_YEAR * 4;
+    private static int NUM_BYTES_PER_FREQ_VALUE = 4;
+    
+    public DateFrequencyValue() {}
+    
+    /**
+     *
+     * @param dateToFrequencyValueMap
+     *            the keys should be dates in yyyyMMdd format
+     * @return Value the value to store in accumulo
+     */
+    public Value serialize(TreeMap<YearMonthDay,Frequency> dateToFrequencyValueMap) {
+        
+        Value serializedMap;
+        int year, presentYear = 0;
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        byte[] dayFrequencies = null;
+        
+        for (Map.Entry<YearMonthDay,Frequency> dateFrequencyEntry : dateToFrequencyValueMap.entrySet()) {
+            year = dateFrequencyEntry.getKey().year;
+            if (year != presentYear) {
+                
+                if (dayFrequencies != null) {
+                    try {
+                        baos.write(dayFrequencies);
+                    } catch (IOException ioException) {
+                        log.error("Did not write out the frequence properly ", ioException);
+                    }
+                }
+                
+                presentYear = year;
+                try {
+                    baos.write(Base256Compression.numToBytes(presentYear));
+                    dayFrequencies = new byte[NUM_FREQUENCY_BYTES];
+                } catch (IOException ioException) {
+                    log.error("Could not convert the year or the first ordinal (julian) to bytes ", ioException);
+                }
+            }
+            
+            putFrequencyBytesInByteArray(dateFrequencyEntry, dayFrequencies);
```

Comment:
Can't this method write the frequencies directly to the output stream ?

Reply:
That's a really great idea.  I will implement that.

Reply:
Well can we at least avoid reallocating a new array every time at line 60 and instead simply 0 our the array: Arrays.fill(dateFrequencies, (byte)0);

Reply:
I finally see where you are going here.  You are really safeguarding against a memory overflow and I have not had to be this careful in past work or I just got lucky.  I will implement your suggestion.  Thanks for working with me here.

Reply:
Array allocation change made and pushed.

Reply:
I really think we could write directly to the baos.  You have to zero out the array anyway so what is the difference?  Just need to write zeros since the last day written when writing the next value.